core/src/main/java/io/questdb/griffin/engine/functions/str/LPadFunctionFactory.java

```diff
@@ -0,0 +1,174 @@
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.griffin.engine.functions.TernaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.Chars;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.Nullable;
+
+public class LPadFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "lpad(SIS)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        final Function strFunc = args.getQuick(0);
+        final Function lenFunc = args.getQuick(1);
+        final Function fillTextFunc = args.getQuick(2);
+        if (fillTextFunc.isNullConstant()) {
+            return new LPadConstFillTextFunc(strFunc, lenFunc);
+        } else {
+            return new LPadFunc(strFunc, lenFunc, fillTextFunc);
+        }
+    }
+
+    public static class LPadConstFillTextFunc extends StrFunction implements BinaryFunction {
+        private final Function strFunc;
+        private final Function lenFunc;
+
+        private final StringSink sink = new StringSink();
+        private final StringSink sinkB = new StringSink();
+
+        public LPadConstFillTextFunc(Function strFunc, Function lenFunc) {
+            this.strFunc = strFunc;
+            this.lenFunc = lenFunc;
+        }
+
+        @Override
+        public Function getLeft() {
+            return strFunc;
+        }
+
+        @Override
+        public Function getRight() {
+            return lenFunc;
+        }
+
+        @Override
+        public int getStrLen(Record rec) {
+            return strFunc.getStrLen(rec);
+        }
+
+        @Override
+        public CharSequence getStr(final Record rec) {
+            return lPad(strFunc.getStr(rec), lenFunc.getInt(rec), sink);
+        }
+
+        @Override
+        public CharSequence getStrB(final Record rec) {
+            return lPad(strFunc.getStr(rec), lenFunc.getInt(rec), sinkB);
+        }
+
+        @Nullable
+        private static StringSink lPad(CharSequence str, int len, StringSink sink) {
+            if (str != null && len != Numbers.INT_NaN && len >= 0) {
+                sink.clear();
+                if (len > str.length()) {
+                    sink.put(Chars.repeat(" ", len - str.length()).toString() + str.toString());
```

Comment:
This line still involves allocations. Please replace it with a for loop with `sink.put(' ')` and a `sink.put(str)` call.